### PR TITLE
Issue #13345: Enable examples tests for InterfaceMemberImpliedModifie…

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for InterfaceMemberImpliedModifierCheck
**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **  nterfaceMemberImpliedModifierCheck** in the project. The implementation focuses on ensuring that   nterfaceMemberImpliedModifier checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+// xdoc section -- start
+class Example1 {
+  public interface AddressFactory {
+
+    public static final String UNKNOWN = "Unknown";
+    // violation below, 'Implied modifier 'final' should be explicit'
+    String OTHER = "Other";
+    // violation above, 'Implied modifier 'public' should be explicit'
+    // violation 2 lines above 'Implied modifier 'static' should be explicit'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java:20]: Please use `// 2 violations above:`
It looks more clearly

**Review by romani — CHANGES_REQUESTED** [review]
Minor

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java (hunk 1 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java]: Done.

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.modifier.interfacememberimpliedmodifier;
+
+import java.util.List;
+
+// xdoc section -- start
+class Example1 {
+  public interface AddressFactory {
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java:14]: Can we avoid wrapper class?
And make 
`public interface Example1 {`

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java (hunk 2 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java:14]: What is a violation?

Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java:14]: ` Example test input must declare exactly one class matching 'Example\d+' [exampleClassName]`

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java:14]: Please update https://github.com/checkstyle/checkstyle/blob/31dbc43ba0e8baf15872c0c62fc379197fb1d419/config/checkstyle-examples-checks.xml#L123

To allow `interface` and `record` and `enum`

Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java]: Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Review by romani — APPROVED** [review]
Awesome

**Pull request merged**